feat: delay loading error-callsites module (#2906)
elastic/apm-agent-nodejs: CHANGELOG.asciidoc

  For example `require('node:http')` will now be instrumented.
  ({issues}2816[#2816])

> - Agent will delay loading of the the `error-callsites` module until agent start time,
>   and will not load the module if the agent is disabled/un-active. This prevents the
>   setting of an Error.prepareCallStack handler and any other side-effects from 
>   `error-callsites` ({issues}2833[#2833] {pull}2906[#2906])

Reviewer: ```suggestion
- Agent will delay loading of the `error-callsites` module until agent start time,
  and will not load the module if the agent is disabled/inactive. This prevents the
  setting of an `Error.prepareStackTrace` handler until necessary for stacktrace
  collection. ({issues}2833[#2833] {pull}2906[#2906])
```
